src/xpk/commands/cluster_gcluster.py

```diff
@@ -0,0 +1,146 @@
+      args.project, zone_to_region(args.zone), args.cluster
+  )
+
+  bp = generate_blueprint(blueprint_name=unique_name, args=args)
+  # staging: sending the blueprint file(s) to gcluster's working directory
+  bp_staged_path = gcm.stage_files(
+      blueprint_file=bp.blueprint_file,
+      blueprint_dependencies=bp.blueprint_dependencies,
+  )
+  gcm.deploy(blueprint_path=bp_staged_path, deployment_name=unique_name)
+
+  xpk_exit(0)
+
+
+def cluster_delete(args) -> None:
+  """Function around cluster delete for the clusters created by Cluster toolkit.
+
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if successful and 1 otherwise.
+  """
+  check_gcloud_authenticated()
+  prepare_directories()
+  gcm = prepare_gcluster_manager()
+  unique_name = get_unique_name(
+      args.project, zone_to_region(args.zone), args.cluster
+  )
+
+  gcm.destroy_deployment(deployment_name=unique_name)
+
+  xpk_exit(0)
+
+
+def created_by_gcluster(args) -> bool:
+  prepare_directories()
+  unique_name = get_unique_name(
+      args.project, zone_to_region(args.zone), args.cluster
+  )
+  bpg = prepare_blueprint_generator()
+  return bpg.blueprint_exists(unique_name)
+
+
+def get_unique_name(project_id, region, cluster_name):
+  unique_string_hash = hash_string(
+      input_string=f'{project_id}-{region}-{cluster_name}'.lower(), length=5
+  )
+  return f'{cluster_name}-{unique_string_hash}'
+
+
+def prepare_directories() -> None:
+  ensure_directory_exists(blueprints_path)
+  ensure_directory_exists(gcluster_working_dir)
+
+
+def check_gcloud_authenticated():
+  if not os.path.exists(gcloud_cfg_path):
+    xpk_print(
+        'Failed to find gcloud credential directory.'
+        f' {gcloud_cfg_path} {blueprints_path} {gcluster_working_dir}'
+    )
+    xpk_print(
+        'Please authenticate to gcloud ("gcloud auth application-default'
+        ' login") and then run your command.'
+    )
+    xpk_exit(-1)
+
+
+def prepare_gcluster_manager() -> GclusterManager:
+  dm = DockerManager(
+      working_dir=gcluster_working_dir, gcloud_cfg_path=gcloud_cfg_path
+  )
+  dm.initialize()
+  return GclusterManager(gcluster_command_runner=dm)
+
+
+def prepare_blueprint_generator() -> BlueprintGenerator:
+  return BlueprintGenerator(storage_path=blueprints_path)
+
+
+def generate_blueprint(blueprint_name, args) -> BlueprintGeneratorOutput:
+  bpg = prepare_blueprint_generator()
+
+  if args.device_type in supported_device_types:
+    if args.device_type == a3mega_device_type:
+      return bpg.generate_a3_mega_blueprint(
+          blueprint_name=blueprint_name,
+          cluster_name=args.cluster,
+          region=zone_to_region(args.zone),
+          project_id=args.project,
+          zone=args.zone,
+          auth_cidr=all_IPs_cidr,
+          num_nodes=args.num_nodes if not args.num_nodes is None else 2,
+          autoscaling_total_min_nodes=args.num_nodes
+          if not args.num_nodes is None
```

Comment:
this line breaking is a little bit unnatural to me, maybe we can set in var before return?